.circleci/config.yml

```
@@ -465,7 +465,7 @@ jobs:
     <<: *build_ubuntu2004
     environment:
       MAKEFLAGS: -j 10
-      CMAKE_OPTIONS: -DCMAKE_BUILD_TYPE=Release -DSOLC_LINK_STATIC=1 -DUSE_CVC4=OFF -DUSE_Z3=OFF
+      CMAKE_OPTIONS: -DCMAKE_BUILD_TYPE=Release -DUSE_Z3_DLOPEN=ON -DUSE_CVC4=OFF -DSOLC_STATIC_STDLIBS=ON
```

```suggestion
      CMAKE_OPTIONS: -DCMAKE_BUILD_TYPE=Release -DUSE_Z3=DLOPEN -DUSE_CVC4=OFF -DSOLC_STATIC_STDLIBS=ON
```
I wonder if CMake would allow something like that (without having to put too much manual effort into it).

---

I actually thought about it - it could be done, but it hides the option away e.g. in ``ccmake`` - so I opted for two options it's easier to see what options one has that way... I could change it, though.

---

Why only for Ubuntu 20.04?

---

Fair enough. My motivation for making it an enum would be that they're all mutually exclusive so you don't have to worry about combinations that do not make sense like `-DUSE_Z3_DLOPEN=ON -DUSE_Z3=OFF`.

---

Yep true. Not sure which of the two points weigh more... so I'd just keep it as is for now, we can still change it later, if we want.